src/platform/qt/library/LibraryController.h

```diff
@@ -0,0 +1,135 @@
+/* Copyright (c) 2014-2017 waddlesplash
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#pragma once
+
+#include <QList>
+#include <QMap>
+#include <QSharedPointer>
+#include <QStackedWidget>
+
+#include <mgba/core/library.h>
+
+namespace QGBA {
+
+// Predefinitions
+class LibraryGrid;
+class LibraryTree;
+class ConfigController;
+
+enum class LibraryStyle {
+	STYLE_LIST,
+	STYLE_TREE,
+	STYLE_GRID,
+	STYLE_ICON
+};
+
+static QString NiceSizeFormat(size_t filesize)
```

Comment:
Where should it go? It's just a size formatter (copied out of your library code). I suppose it could go on just the tree. In DolphinQt, this was in a Utils.h.